tools/tgs_scripts/InstallDeps.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+#find out what we have (+e is important for this)
+set +e
+has_git="$(command -v git)"
+has_cargo="$(command -v ~/.cargo/bin/cargo)"
+has_sudo="$(command -v sudo)"
+has_youtubedl="$(command -v youtube-dl)"
+has_pip3="$(command -v pip3)"
+set -e
+set -x
+
+# install cargo if needed
+if ! [ -x "$has_cargo" ]; then
+	echo "Installing rust..."
+	curl https://sh.rustup.rs -sSf | sh -s -- -y
+	. ~/.profile
+fi
+
+# apt packages, libssl needed by rust-g but not included in TGS barebones install
+if ! ( [ -x "$has_git" ] && [ -f "/usr/lib/i386-linux-gnu/libssl.so" ] ); then
+	echo "Installing apt dependencies..."
+	if ! [ -x "$has_sudo" ]; then
+		dpkg --add-architecture i386
+		apt-get update
+		apt-get install -y lib32z1 git pkg-config libssl-dev:i386 libssl-dev libssl1.1:i386
```

Comment:
libssl1.1 doesn't exist anymore on Ubuntu 22.04. TGS still requires (the 64 bit version of) it, but there's workarounds for that. In practice I've successfully been able to compile rustg without it, presumably it can just use libssl3?